q2_keegan/vanilla/_methods.py

```diff
@@ -0,0 +1,83 @@
+import csv
+
+from . import VanillaBeanFmt, IceCream
+from qiime2.plugin import Str, Choices, Bool, TypeMap
+
+from ..plugin_setup import plugin
+
+
+def syrup() -> VanillaBeanFmt:
+
+    ff = VanillaBeanFmt()
+
+    with ff.open() as fh:
+        pour = csv.writer(fh, delimiter="\t")
+        pour.writerow(['Chocolate Fudge', ])
+
+    return ff
+
+
+plugin.methods.register_function(
+    function=syrup,
+    inputs={},
+    parameters={},
+    outputs=[('cute_sent', IceCream)],
+    name='syrup',
+    description='A tiny proof of concept function'
+)
+
+
+def sprinkle(icecream: VanillaBeanFmt,
+             capitalization: str,
+             sort_ascending: bool = False,
+             sort_descending: bool = False) -> list:
+
+    contents = []
+
+    with icecream.open() as fh:
+        for line in fh:
+            contents.append(line)
+
+    # Sort if appropriate
+    sort_var = None
+    if sort_ascending:
+        sort_var = False
+    if sort_descending:
+        sort_var = True
+    if sort_var is not None:
+        contents.sort(key=lambda x: x[0].lower(), reverse=sort_var)
```

Comment:
Please apply this optional sorting step after the mandatory transformation step.

Reply:
Also, I don't think this sorting algorithm is quite right - sorting should consider the whole line's contents, and should honor the line's casing.